### PR TITLE
🌱 Add Pinned-Dependency, Vulnerability, and Code-Review checks to attestor

### PR DESCRIPTION
**Pull request by raghavkaul**
#### What kind of change does this PR introduce?
Add more checks to the scorecard attestor. Also clean up some lint. 

All checks work on Raw Results

- [x] Tests for the changes have been added (for bug fixes/features)

#### Does this PR introduce a user-facing change?

```release-note
Scorecard-Attestor now supports attestations based on the Pinned-Dependency, Vulnerability, and Code-Review checks
```

### CODE REVIEW

File: attestor/policy/attestation_policy.go
Comment (spencerschrock) [inline attestor/policy/attestation_policy.go:251]: nit: follow-up of nit above. 
You could just iterate over `reqs.RequiredApprovers` instead of the entire `approvers` map.

File: attestor/policy/attestation_policy.go
Comment (raghavkaul) [reply attestor/policy/attestation_policy.go]: I like that better! Changed.

File: attestor/policy/attestation_policy.go
Comment (raghavkaul) [reply attestor/policy/attestation_policy.go]: Good point, changed this.

**Review by spencerschrock — APPROVED** [review]
Thanks! 

I know this is blocking the `attestor` module -> package PR, but once that is merged can you add some additional tests which cover `EvaluateResults` and `GetRequiredChecksForPolicy`. Maybe a few e2e tests on test repos?

**Comment by raghavkaul** [comment]
> Thanks!
> 
> I know this is blocking the `attestor` module -> package PR, but once that is merged can you add some additional tests which cover `EvaluateResults` and `GetRequiredChecksForPolicy`. Maybe a few e2e tests on test repos?

Sure thing! It will be easier to use ginkgo too once this is in a subpackage.

**Pull request merged**
